Develop

barhenkro opened the pull request.

barhenkro committed:
updated values

barhenkro committed:
general fixes

barhenkro committed:
fixed registry problems

barhenkro committed:
added test script

barhenkro committed:
general fixes

barhenkro committed:
Merge branch 'camera' of https://github.com/TheJoker4320/vision-framework into test

barhenkro committed:
merged camera into the project

barhenkro committed:
general fixes

barhenkro committed:
PEP8 fixes

barhenkro committed:
fixed conflicts

barhenkro committed:
Delete __init__.pyc

barhenkro committed:
Delete blur.pyc

barhenkro committed:
Delete color_threshold.pyc

barhenkro committed:
updated json file

barhenkro committed:
changed setting setter to static method

barhenkro committed:
added use of camera to main

barhenkro committed:
changed logs to debug level

barhenkro committed:
added camera id to json file

barhenkro committed:
removed debugging printing

barhenkro committed:
added logger publisher

barhenkro committed:
general fixes

barhenkro committed:
fixed setter function

barhenkro committed:
fixed empty list problem

barhenkro committed:
updtaed values

barhenkro committed:
added gui tests in comments

barhenkro committed:
added gui test in comments

LilApple committed:
PEP8 fixes

LilApple committed:
pr fixes

barhenkro committed:
PR fixes

barhenkro committed:
Merge branch 'develop' into test

barhenkro committed:
PEP8 fixes

barhenkro committed:
Merge pull request #10 from TheJoker4320/test

Test

LilApple committed:
Rename package_utilis.py to package_utils.py

LilApple commented on package_utilis.py:
```diff
@@ -0,0 +1,30 @@
+import os
```
Changed

tzmoz123 commented on camera.py:
```diff
@@ -0,0 +1,184 @@
+        """
+        self.port = port
+        self.video_capture = cv2.VideoCapture(self.port)
+        self.initial_string = "{template}{port}".format(template=Camera.template, port=self.port)
```
That's how it works

tzmoz123 committed:
general fixes

LilApple commented on package_utilis.py:
```diff
@@ -0,0 +1,30 @@
+    :return: True or false
+    :rtype boolean
+    """
+    if file_name == 'modifier.py':
```
Resolved

LilApple committed:
Removed commented lines

LilApple committed:
Added blacklist

LilApple commented on main.py:
```diff
@@ -0,0 +1,27 @@
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+    # cv2.namedWindow("screen1")
```
Removed

LilApple replied:
Removed

LilApple commented on main.py:
```diff
@@ -0,0 +1,27 @@
+from pipeline.pipeline_factory import PipelineFactory
+import logging
+from camera import Camera
+import cv2
```
Removed

LilApple replied:
Removed

LilApple committed:
General fixes

LilApple committed:
Removed decortors

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+        return False
+
+    @staticmethod
+    def modifier(name):
```
Removed

LilApple committed:
General fixes

LilApple committed:
Rename utils.py to calculation_utils.py

tzmoz123 committed:
general fixes

tzmoz123 committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

barhenkro commented on filters/area_ratio_filter.py:
```diff
@@ -0,0 +1,38 @@
+        point1, point2, point3, _ = cv2.boxPoints(rect)
+        rect_height = utils.distance(point1, point2)
+        rect_width = utils.distance(point2, point3)
+        rect_area = rect_width * rect_height
```
The function cv2.boxPoints returns the 4 points of the bounding rectangle ,so it will always be the rectangle that bounds the contour

LilApple committed:
General fixes

tzmoz123 committed:
general fixes

tzmoz123 committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

tzmoz123 committed:
general fixes

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+from pipeline import Pipeline
+import logging
+from modifiers import *
```
Resolved

LilApple committed:
Removed ugly imports

tzmoz123 committed:
general fixes

tzmoz123 committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

LilApple committed:
Moved props to examples and change its name to example

LilApple committed:
General fixes

LilApple committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

LilApple commented on properties.json:
```diff
@@ -0,0 +1,85 @@
+{
```
Fixed

tzmoz123 committed:
general fixes

LilApple commented on utils.py:
```diff
@@ -0,0 +1,17 @@
+import math
```
Fixed

tzmoz123 committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+
+    @staticmethod
+    def create_pipeline(proprties):
+        modifier_dict = PipelineFactory.get_modifiers(proprties)
```
Fixed

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+        :return: list of build instances from the same type that was recieved
+        :rtype: list
+        """
+        return [registry[name](**properties) for name, properties in dictionary.iteritems() if
```
Thx :)

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+                PipelineFactory.__in_registry(name, registry)]
+
+    @staticmethod
+    def __in_registry(name, registry):
```
Fixed

LilApple commented on pipeline/pipeline.py:
```diff
@@ -0,0 +1,80 @@
+        # cv2.waitKey(30)
+        # cv2.imshow("screen2",frame)
+        """
+        this iteration responsible for publishing via different publishers
```
Removed, those comments were for debugging

LilApple commented on pipeline/pipeline_factory.py:
```diff
@@ -0,0 +1,145 @@
+
+    @staticmethod
+    def create_pipeline(proprties):
+        modifier_dict = PipelineFactory.get_modifiers(proprties)
```
Fixed

LilApple committed:
General fixes

LilApple committed:
Removed comments and raised warning level

LilApple commented on pipeline/pipeline.py:
```diff
@@ -0,0 +1,80 @@
+        """
+        contour_amount = len(contours)
+        if contour_amount == 0:
+            logging.debug("there were not found any counter")
```
Done

LilApple committed:
Changed names

LilApple committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into develop

tzmoz123 committed:
Update and rename logger.py to logging_publisher.py

general fixes

LilApple commented on pipeline/pipeline.py:
```diff
@@ -0,0 +1,80 @@
+        """
+        for calculation in self.calculations:
+            for publisher in self.publishers:
+                publisher.publish(calculation.calc(contour))
```
Fixed

LilApple committed:
General fixes

LilApple committed:
Guy fixes

LilApple commented on requirements.txt:
```diff
@@ -1 +1,2 @@
+pynetworktables
```
Y

tzmoz123 replied:
E

barhenkro replied:
S

LilApple committed:
fixes

LilApple committed:
general fixes

barhenkro commented on package_utilis.py:
```diff
@@ -0,0 +1,30 @@
+    :return: True or false
+    :rtype boolean
+    """
+    if file_name == 'modifier.py':
```
an explanation for modifier for instance:
in pipeline factory we need to use the Modifier class to get his children.
to do so we need to import modifier.py but importing this with the whole package will raise an error because it will not recognize the Modifier class.
that is why we excluded those files because including them won't give any use

barhenkro committed:
added ordered dictionery to the properties loading

barhenkro committed:
created new filter

barhenkro committed:
added consideration to multiply contours

barhenkro committed:
added blank function to merge contours

barhenkro committed:
finished merge contours function

barhenkro committed:
PEP8 fixes

barhenkro committed:
added a streamer class

barhenkro committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into streamer

barhenkro committed:
added streamer class

barhenkro committed:
Finished streamer and used his functions

barhenkro committed:
Made the process image to return processed image

barhenkro committed:
Bar-hen fixes

barhenkro committed:
workshop fixes

barhenkro committed:
Started remote values configuration

LilApple committed:
added flask to requirements

LilApple committed:
test version

barhenkro committed:
finished and tested streamer

barhenkro committed:
startedd building RemoteTuner class

barhenkro committed:
added listener

barhenkro committed:
Renamed file

barhenkro committed:
Merge branch 'develop' of https://github.com/TheJoker4320/vision-framework into diagonal-reflective-tape

barhenkro committed:
merged streamer with the diagonal reflective tape filter

barhenkro committed:
Finished and tested diagonal reflective tape pair filter

barhenkro committed:
added extractors to the framework

barhenkro committed:
added extractor that extracts circles

Commit:
convert int to string in dict

barhenkro committed:
general fixes

barhenkro committed:
added utils for numpy

barhenkro committed:
fix bugs temp

Commit:
workshop fixes

LilApple committed:
Added new calculation

LilApple committed:
Made calcs work with multiple contours and made pipline work with them

LilApple committed:
general fixes

barhenkro committed:
bug fixes

barhenkro committed:
added ball examples

tzmoz123 committed:
add turn calculation values to json file

barhenkro committed:
bug fixes

tzmoz123 committed:
general fixes

tzmoz123 committed:
general fixes

barhenkro committed:
fixes due to cahnges in calculation class

barhenkro committed:
Merge branch 'diagonal-reflective-tape' of https://github.com/TheJoker4320/vision-framework into diagonal-reflective-tape

barhenkro committed:
general fixes

Commit:
general fixes

barhenkro committed:
merged tuning with circle detection

barhenkro committed:
Jetson fixes

barhenkro committed:
Workshop fixes

barhenkro committed:
Workshop fixes

LilApple committed:
Workshop Fixes

LilApple committed:
Fixed PEP8 Problems

LilApple committed:
Fixed PEP8 problems

LilApple committed:
Fixed PEP8 problems

LilApple committed:
Merged and Fixed the conflicts

LilApple committed:
Merged and Fixed the conflicts

LilApple committed:
Merged and Fixed the conflicts

tzmoz123 committed:
adjusting the main for several pipelines

tzmoz123 committed:
fixed merge conflict

LilApple committed:
Begining of PEP8 Fixes

LilApple committed:
Begining of PEP8 Fixes

LilApple committed:
Begining of PEP8 Fixes

barhenkro committed:
workshop fixes

barhenkro committed:
workshop fixes

barhenkro committed:
added json file

LilApple committed:
Preapring to move to py3

LilApple committed:
Circlei Fixes

LilApple committed:
general fixes

LilApple committed:
general fixes

LilApple committed:
Circlei Fixes

barhenkro committed:
Added the api refrence option

barhenkro committed:
Fixed the contour find error

barhenkro committed:
Reworked the entire main file

barhenkro committed:
Fully converted to py3

barhenkro committed:
Fixed the area range filter

LilApple committed:
Created the new distance calculation

LilApple committed:
Fixed the import issue

LilApple committed:
Renamed from distance_calculation

LilApple committed:
Fixed main PEP8 issues

tzmoz123 committed:
general fixes

tzmoz123 committed:
add file for converting GRIP file to our JSON file

tzmoz123 committed:
add GRIP file for testing

tzmoz123 committed:
fixes according to the pull request

barhenkro committed:
updated readme file

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

tzmoz123 committed:
general fixes and calculation fixes

tzmoz123 committed:
change name of calc_constant to distance_function

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

tzmoz123 committed:
add self to distance_function

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

tzmoz123 committed:
add new calc according to vector

LilApple committed:
Update README.md

LilApple committed:
Update README.md

LilApple committed:
Update README.md

tzmoz123 committed:
fix the calculation and add documentation

tzmoz123 committed:
fix the main to work

tzmoz123 committed:
move 'create_rotation_matrix' function to calculation utils

tzmoz123 committed:
update python version

tzmoz123 committed:
remove unwanted whitespace

tzmoz123 committed:
Merge branch 'distance-calculation-by-vector' of https://github.com/TheJoker4320/vision-framework into distance-calculation-by-vector

tzmoz123 committed:
update python version

tzmoz123 committed:
add documentation and rename file

tzmoz123 committed:
set a local variable for the eval function

tzmoz123 committed:
fix the main to work

tzmoz123 committed:
remove unwanted whitespace

tzmoz123 committed:
edit the json file

change the name of the calculation (the class name was updated) and change the symbol of power from ^ to **

tzmoz123 committed:
general fixes for the main

tzmoz123 committed:
Merge remote-tracking branch 'origin/develop' into new_calc

tzmoz123 committed:
rename file

from  "convertGRIPtoJSON.py" to "convert_grip_to_json.py"

tzmoz123 committed:
edit the convertor

add documentation, organized the code flow and add content to functions

tzmoz123 committed:
fix the convertor

fix the function that call the edit functions to work with the dictionary

tzmoz123 committed:
update python version

tzmoz123 committed:
fix the main to work

tzmoz123 committed:
fix the main to work

remove unwanted whitespace

tzmoz123 committed:
remove unwanted whitespace

tzmoz123 committed:
preparing for merge develop in grip-to-json

change the main

tzmoz123 committed:
Merge remote-tracking branch 'origin/develop' into grip-to-json

tzmoz123 committed:
Merge pull request #17 from TheJoker4320/new_calc

New calc

tzmoz123 committed:
Merge branch 'develop' of github.com:TheJoker4320/vision-framework into grip-to-json

tzmoz123 committed:
edit the "convert_grip_to_json.py" file

add documentation and fix naming

tzmoz123 committed:
edit documentation

tzmoz123 committed:
remove unwanted whitespaces

tzmoz123 committed:
Merge pull request #14 from TheJoker4320/grip-to-json

Grip to json

tzmoz123 committed:
update python version

tzmoz123 committed:
remove unwanted whitespace

tzmoz123 committed:
fix main to work

tzmoz123 committed:
fix the main to work

tzmoz123 committed:
fix spelling errors

tzmoz123 committed:
test Table of Contents

try to create a Table of Contents to the readme file

tzmoz123 committed:
add Table of Contents

tzmoz123 committed:
add grip file support data

tzmoz123 committed:
add data about the Pipeline Creation

add all the available options (for now) in the framework and their implement in the JSON file

tzmoz123 committed:
add punctuation

tzmoz123 committed:
add links to tables and general fixes

tzmoz123 committed:
update the readme file

edit guide for installation, add streamer, remote tuner and how to tun the framework

tzmoz123 committed:
add data about the camera setting

tzmoz123 committed:
add data about the streamer

tzmoz123 committed:
edit the Table of Contents

add the streamer, remote tuner and run the framework headlines

tzmoz123 committed:
add remote tuner data

tzmoz123 committed:
Merge branch 'develop' of github.com:TheJoker4320/vision-framework into documentaion

tzmoz123 committed:
Merge pull request #15 from TheJoker4320/documentaion

updated readme file
